dimod/discrete/discrete_quadratic_model.py

```diff
@@ -810,3 +808,315 @@ def to_numpy_vectors(self, return_offset: bool = False):
+        self._unique_label_case = {}
+
+    def add_variable(self, cases, label=None, shared_labels=False):
+        """Add a discrete variable to the model.
+
+        Args:
+            cases (int or iterable):
+                The number of cases in the variable, or an iterable containing
+                the labels that will identify the cases of the variable.
+
+            label (hashable, optional):
+                A label for the variable. Can be any hashable except `None`.
+                Defaults to the length of the discrete quadratic model, if that
+                label is available. Otherwise defaults to the lowest available
+                positive integer label.
+
+            shared_labels (bool, optional, default=False):
+                If True and `cases` is an iterable, shared case labels are
+                created.  If False and `cases` is an iterable, unique case
+                labels are created.  If `cases` is not an iterable, ignored.
+
+        Returns:
+            The label of the new variable.
+
+        Raises:
+            ValueError: If `label` already exists as a variable label, or if
+                any of the case labels is not unique.
+
+            TypeError: If `label` is not hashable, or if any of the case labels
+                is not hashable.
+
+        """
+        if label in self._unique_label_case:
+            raise ValueError(f'variable label {label} is not unique')
+
+        if isinstance(cases, int):
+            return super().add_variable(cases, label=label)
+
+        else:
+            if len(set(cases)) != len(cases):
+                raise ValueError('case labels are not unique')
+
+            if shared_labels:
+                var = super().add_variable(len(cases), label=label)
+
+                for k, case in enumerate(cases):
+                    self._shared_label_case[var][case] = k
+                    self._shared_case_label[var][k] = case
+
+            else:
+                for case in cases:
+                    if (case in self.variables) or (case in self._unique_label_case):
+                        raise ValueError(f'case label {case} is not unique')
+
+                var = super().add_variable(len(cases), label=label)
+
+                for k, case in enumerate(cases):
+                    self._unique_label_case[case] = (var, k)
+                    self._unique_case_label[(var, k)] = case
+
+            return var
+
+    def _lookup_shared_case(self, v, case):
+        """Translate shared case label `case` of variable `v` to integer.
+
+        Raises:
+            ValueError: If `case` of `v` is unknown.
+
+        """
+        map_ = self._shared_label_case.get(v)
+        if map_:
+            if case not in map_:
+                raise ValueError(f'unknown case {case} of variable {v}')
+            return map_[case]
+        return case
+
+    def get_linear(self, v):
+        """The linear bias(es) associated with variable `v`.
```

Comment:
"The linear biases associated ... " is more readable and no one will hold the one-case corner case against you .
(Also in the returns description here and some other places)

Reply:
Done.